### ISSUE TITLE
std::ios_base::floatfield misused in jsontest.h

### ISSUE THREAD

**Issue by BillyDonahue** [open-source-parsers/jsoncpp#1078]
TestResult operator<< uses

    oss.setf(std::ios_base::floatfield);

This is inappropriate. The ios_base::floatfield is a mask, not a value.
This essentially selects hexfloat output, but if we wanted to do that we should do it explicitly.

**Reproduce**

This essentially what our TestResult operator is doing:

https://gcc.godbolt.org/z/e93L45

```
#include <iostream>
int main() {
    std::cout.setf(std::ios_base::floatfield);
    std::cout << "f:" << 3.14 << std::endl;
}
Output:
f:0x1.91eb851eb851fp+1

```

**Expected behavior**
I think we didn't want hexfloat output, or we would have asked for it by name.
I'm guessing the intent was to clear the floatfield, and we do that with `unsetf`.

None of our unit tests actually call operator<< with a floating point value, so it never mattered. We should fix the bug anyway.

**Comment by cdunn2001** [comment]
`floatfield` came from:
```
commit f587e6a420c467a69b7f977875052077aef2913b
Author: Baptiste Lepilleur <baptiste.lepilleur@gmail.com>
Date:   Thu May 26 15:55:24 2011

    Fixed compilation issues with MSVC 6: replace usage of ostringstream with valueToString to support 64 bits integer and high precision floating point conversion to string.
```
I guess nobody else ever thought to check it! You can close this, or leave it open to try `<<` with a float.

**Comment by BillyDonahue** [comment]
fixed by 638ad269e75f28f9830f7bfc01278aeaf5b00135
